### DISCUSSION TITLE
Trying to setup Draco

### DISCUSSION THREAD

**Discussion by EdwardMoyse** [HSF/phoenix#529] [Q&A] [answered]
This is probably a question for @9inpachi ;-)

I'm trying to get DRACOLoader working, by adding:
```
    const loader = new GLTFLoader();
    const dracoLoader = new DRACOLoader();
    dracoLoader.setDecoderPath('three/examples/js/libs/draco/');
```
Obviously I've not managed, hence this discussion!

The error I get is that `draco_wasm_wrapper.js` is not found.
```
Summary
URL: http://localhost:4200/three/examples/js/libs/draco/draco_wasm_wrapper.js
Status: 404 Not Found
Source: Network
Address: 127.0.0.1:4200
Initiator: 
zone.js:1518
```

However it's here: `node_modules/three/examples/js/libs/draco`

```
emoyse@Edwards-MacBook-Pro phoenix % ls node_modules/three/examples/js/libs/draco 
README.md               draco_decoder.wasm      draco_wasm_wrapper.js
draco_decoder.js        draco_encoder.js        gltf
```

Any idea what I'm doing wrong? I was wondering if I need to declare that I need draco somewhere so it is available to the web app?

**Comment by 9inpachi** [comment, marked as answer]
The loader is trying to load the file from the server but the server is not serving this file since it's an npm dependency and only exists in `node_modules`. Our options here are:

1. Put the `DRACOLoader` JS files in the app's `assets` folder and specify the path from assets: `dracoLoader.setDecoderPath('assets/js/draco-loader/');`
2. Use the URL from jsDelivr: `dracoLoader.setDecoderPath('https://cdn.jsdelivr.net/npm/three@0.147.0/examples/js/libs/draco/');`

There might be some other better options I don't know of.

**Reply by EdwardMoyse** [reply]
Thanks @9inpachi - so in either case, the issue would be that when we update threejs version we need to remember to either update `DRACOloader` or the threejs version in the URL? If so, I think maybe I slightly prefer the second option (since it seems better to avoid adding unnecessary copied code to our repository).

**Reply by 9inpachi** [reply]
The 2nd option is better for us. And we should be able to get the current three.js version and use it in the jsDelivr URL:

```ts
import { REVISION } from 'three';

dracoLoader.setDecoderPath(`https://cdn.jsdelivr.net/npm/three@0.${REVISION}.0/examples/js/libs/draco/`);
```

**Reply by EdwardMoyse** [reply]
Thanks @9inpachi  - as you can see from #530, I implemented this now.
